setup.py

```diff
@@ -25,8 +33,18 @@
     long_description = f.read()
 
 
-# the extension is optional since in case of lack of c the api
-# there is a ctypes fallback and a slow python fallback
+class BuildExt(base_build_ext):
+
+    # the extension is optional since in case of lack of c the api
+    # there is a ctypes fallback and a slow python fallback
+
+    def build_extension(self, ext):
+        try:
+            base_build_ext.build_extension(self, ext)
+        except Exception:
+            exception = sys.exc_info()[0]
+            logging.warn("building the %s failed with %s", ext.name, exception)
+
 extension = Extension('_scandir', ['_scandir.c'], optional=True)
```

Comment:
Won't this still fail as we still always include the `optional=True` flag here? Or is it just a warning and so we're ignoring it?

Reply:
its just a warning on python2, i'm ignoring it here

the `build_ext command is what's interpreting those